### PR TITLE
doc: add Pod Security Standards best practices

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
doc: add Pod Security Standards best practices

### CODE REVIEW

File: website/content/en/docs/best-practices/pod-security-standards.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+```yaml
+  labels:
+    ...
+    pod-security.kubernetes.io/enforce: privileged
+    pod-security.kubernetes.io/audit: privileged
+    pod-security.kubernetes.io/warn: privileged
+```
+
+**Note that you should ensure the above configuration is carried to the Pod/Containers on the bundle CSV (install.spec.deployments.containers).**
+To check an example of CSV which comply with the [restrictive][restricted] policy, see the Golang sample
```
Comment (camilamacedo86) [inline website/content/en/docs/best-practices/pod-security-standards.md:101]:
```suggestion
To check an example of CSV which complies with the [restrictive][restricted] policy, see the Golang sample
```

File: website/content/en/docs/best-practices/pod-security-standards.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+## Overview
+
+Kubernetes API has been changing, and the [PodSecurityPolicy][pod-security] API is deprecated and will no longer be served from k8s `1.25`. 
+This API is replaced by a new built-in admission controller ([KEP-2579: Pod Security Admission Control][2579-psp-replacement]) which allows cluster admins to [enforce 
+the Pod Security Standards][enforce-standards-namespace-labels].
+
+#### What does that mean?
+
+That means that Pod/containers that are not configured according to the enforced security standards defined globally or 
+on the namespace level will not be admitted and in this way, it will not be possible to run them.
```
Comment (michaelryanpeter) [inline website/content/en/docs/best-practices/pod-security-standards.md:17]:
```suggestion
You must configure Pods and containers accroding to the Pod Security Standards at a global or namespace level. Pods and containers that do not configured this way will not run.
```

File: website/content/en/docs/best-practices/pod-security-standards.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+**In this way, the best approach is to ensure that workloads (Operators, Operands) are defined such that they can run under restricted permissions.**
```
Comment (michaelryanpeter) [inline website/content/en/docs/best-practices/pod-security-standards.md:19]:
```suggestion
**As a best practice, you must ensure that workloads such as Operators and Operands are defined to run under restricted permissions.**
```

File: website/content/en/docs/best-practices/pod-security-standards.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+#### How should I configure my Operators and Operands to comply with the criteria?
+
+- A) **(Recommend for common cases, when it does not require escalating privileges)** ensure all containers are configured 
+to comply with the [restrictive][restricted] policy. Following the examples:
```
Comment (michaelryanpeter) [inline website/content/en/docs/best-practices/pod-security-standards.md:24]:
```suggestion
- **For common cases that do not require escalating privileges:** configure all containers to comply with the [restrictive][restricted] policy as shown in the following the examples:
```

File: website/content/en/docs/best-practices/pod-security-standards.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      - name: controller-manager
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - ALL
+        ...
+```
+
+**On Reconciliations (i.e of code implementation in Go):**
```
Comment (michaelryanpeter) [inline website/content/en/docs/best-practices/pod-security-standards.md:45]:
```suggestion
**On Reconciliations, such as code implementation in Go:**
```

File: website/content/en/docs/best-practices/pod-security-standards.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+              },
+           }},
+        },
+     },
+  },
+}
+```
+
+**Note:** For Ansible/Helm based-language operators, you need to ensure that your ansible playbooks or charts respectively 
+are creating the manifests complying with the requirements.
```
Comment (michaelryanpeter) [inline website/content/en/docs/best-practices/pod-security-standards.md:85]:
```suggestion
**Note:** For Ansible- and Helm-based Operator projects, your Ansible playbooks or Helm charts must create manifests that comply with the requirements.
```

File: website/content/en/docs/best-practices/pod-security-standards.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+**OR**
+
+- B) Ensure the namespace has the appropriate enforcement level label if the workload needs more than restricted permissions ( check the following example ).
```
Comment (michaelryanpeter) [inline website/content/en/docs/best-practices/pod-security-standards.md:89]:
```suggestion
- B) **For workloads that need elevated permissions:** Ensure the namespace has the appropriate enforcement level label as shown in the following example.
```

**Review by jmrodri — APPROVED** [review]
/lgtm

**Review by laxmikantbpandhare — APPROVED** [review]
/lgtm

File: website/content/en/docs/best-practices/pod-security-standards.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+}
+```
+
+**Note:** For Ansible/Helm based-language operators, you need to ensure that your ansible playbooks or charts respectively 
+are creating the manifests complying with the requirements.
+
+**OR**
+
+- B) Ensure the namespace has the appropriate enforcement level label if the workload needs more than restricted permissions ( check the following example ). 
+This may need to be part of your operator install instructions.  While the label syncer should handle this for you in most cases, it is preferred that the Operator be explicit about its requirements.
```
Comment (michaelryanpeter) [inline website/content/en/docs/best-practices/pod-security-standards.md:90]:
```suggestion
You might need include this in the installation documentation for your Operator.  While the label syncer should handle this for you in most cases, it is a good practice for Operators to explicitly state its requirements.
```

File: website/content/en/docs/best-practices/pod-security-standards.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+```yaml
+  labels:
+    ...
+    pod-security.kubernetes.io/enforce: privileged
+    pod-security.kubernetes.io/audit: privileged
+    pod-security.kubernetes.io/warn: privileged
+```
+
+**Note that you should ensure the above configuration is carried to the Pod/Containers on the bundle CSV (install.spec.deployments.containers).**
```
Comment (michaelryanpeter) [inline website/content/en/docs/best-practices/pod-security-standards.md:100]:
```suggestion
**You should ensure the configuration is carried to the Pod/Containers on the bundle CSV (install.spec.deployments.containers).**
```

File: website/content/en/docs/best-practices/pod-security-standards.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+To check an example of CSV which complies with the [restrictive][restricted] policy, see the Golang sample
+under the [testdata/go/v3/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml](https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/go/v3/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml)
```
Comment (michaelryanpeter) [inline website/content/en/docs/best-practices/pod-security-standards.md:102]:
```suggestion
## Additional resources

- See the Golang sample under the [testdata/go/v3/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml](https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/go/v3/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml) for an example CSV that complies with the [restrictive][restricted] policy.
```

File: website/content/en/docs/best-practices/pod-security-standards.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+[pod-security]: https://kubernetes.io/blog/2021/04/06/podsecuritypolicy-deprecation-past-present-and-future/#what-is-podsecuritypolicy
+[2579-psp-replacement]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/2579-psp-replacement
+[enforce-standards-namespace-labels]: https://kubernetes.io/docs/tasks/configure-pod-container/enforce-standards-namespace-labels/
+[restricted]: https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted
```
Comment (michaelryanpeter) [inline website/content/en/docs/best-practices/pod-security-standards.md:107]:
```suggestion
- [pod-security]: https://kubernetes.io/blog/2021/04/06/podsecuritypolicy-deprecation-past-present-and-future/#what-is-podsecuritypolicy
- [2579-psp-replacement]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/2579-psp-replacement
- [enforce-standards-namespace-labels]: https://kubernetes.io/docs/tasks/configure-pod-container/enforce-standards-namespace-labels/
- [restricted]: https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted
```

**Review by michaelryanpeter — COMMENTED** [review]
I left some suggestions. Please take what is helpful and disregard the rest. Let me know if you have questions.

**Pull request merged**